lnwallet/interface_test.go

```diff
@@ -2533,6 +2533,80 @@ func testLastUnusedAddr(miner *rpctest.Harness,
 	}
 }
 
+func testListWalletAddresses(miner *rpctest.Harness,
+	alice, bob *lnwallet.LightningWallet, t *testing.T) {
+
+	if _, err := miner.Node.Generate(1); err != nil {
+		t.Fatalf("unable to generate block: %v", err)
+	}
+
+	// We'll repeat this test for each address type to ensure they're all
+	// rotated properly.
+	addrTypes := []lnwallet.AddressType{
+		lnwallet.WitnessPubKey, lnwallet.NestedWitnessPubKey,
+	}
+
+	for _, addrType := range addrTypes {
+		addr1, err := alice.LastUnusedAddress(addrType)
+		if err != nil {
+			t.Fatalf("unable to get addr: %v", err)
+		}
+
+		addrList1, err := alice.ListWalletAddresses(addrType)
+		if err != nil {
```

Comment:
nit: we've started using the `require` package in unit tests, `require.NoError` will be useful in a few places here

Reply:
Can you describe an example behavior for this? I'll happily implement it, just need to see how

Reply:
Ah, got it! It'll be in an incoming commit.